### PR TITLE
nf-core smrnaseq release 1.0

### PR DESCRIPTION
**Pull request by chuan-wang**
* Refactors to nf-core smrnaseq;
* Option for adding sequencing center to bam files from aligner;
* Switch to Bowtie 1 for the alignment to host reference genome.

### CODE REVIEW

File: conf/uppmax-modules.config [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+ * -------------------------------------------------
+ *  Nextflow config file with environment modules for UPPMAX
+ * -------------------------------------------------
+ */
+
+singularity {
+  enabled = false
+}
+
+process {
```
Comment (apeltzer) [inline conf/uppmax-modules.config]: This still has the old process$ syntax you replaced further up in the other configs - maybe that would be good here too?

File: conf/uppmax-modules.config [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  // Environment modules and resource requirements
+  $makeBowtieIndex.module = ['bioinfo-tools', 'Fastx/0.0.14', 'bowtie/1.2.0']
```
Comment (apeltzer) [inline conf/uppmax-modules.config]: Same here - process$ instead of withName:

File: conf/uppmax.config [hunk 3]
```diff
@@ -13,6 +13,7 @@
 process {
   executor = 'slurm'
   clusterOptions = { "-A $params.project ${params.clusterOptions ?: ''}" }
+  container = "docker://$wf_container"
```
Comment (apeltzer) [inline conf/uppmax.config]: You don#t need the docker prefix anymore. 
And you could have something like in the base profile here: https://github.com/nf-core/rnaseq/blob/2cf2f0f779d39463b09d874449cbb9fdb9f58d2f/conf/base.config#L14 
inside the profile here too - initializing the variable in nextflow.config https://github.com/nf-core/rnaseq/blob/2cf2f0f779d39463b09d874449cbb9fdb9f58d2f/nextflow.config#L15

**Comment by chuan-wang** [comment]
This PR closes #10

File: main.nf [hunk 4]
```diff
@@ -423,50 +454,57 @@ process edgeR_miRBase {
...
             -p 8 \\
             -t \\
-            | samtools view -bT $index_base - > ${prefix}.bowtie2.bam
+            -k 10 \\
+            -m 1 \\
+            --best \\
+            --strata \\
+            -e 99999 \\
+            --chunkmbs 2048 \\
+            $seqCenter \\
```
Comment (ewels) [inline main.nf]: As above with the newlines

**Comment by ewels** [comment]
Final thing @chuan-wang - please could you update the conda tools to their latest version? This is something we should do for each release:

* trim-galore=0.4.5, 0.5.0 available
* samtools=1.8, 1.9 available
* multiqc=1.5, 1.6 available

(please mention in the changelog)

**Comment by chuan-wang** [comment]
Hi, the pipeline only ran with the parameters of reference in `main.nf`.

**Comment by apeltzer** [comment]
Hmm, I'll have a look later. Will land in a couple of minutes and loose internet connection therefore...

File: nextflow.config [hunk 5]
```diff
@@ -10,34 +10,31 @@
...
-  params.readPaths = null
-  params.outdir = './results'
-  multiqc_config = "$baseDir/conf/multiqc_config.yaml"
-  params.email = false
-  params.plaintext_email = false
-  params.skip_qc = false
-  params.skip_fastqc = false
-  params.skip_multiqc = false
+  name = false
+  project = false
```
Comment (ewels) [inline nextflow.config]: Looks like `singleEnd` has gone missing here.

File: main.nf [hunk 6]
```diff
@@ -313,6 +340,7 @@ process bowtie_miRBase_mature {
     script:
     index_base = index.toString().tokenize(' ')[0].tokenize('.')[0]
     prefix = reads.toString() - ~/(.R1)?(_R1)?(_trimmed)?(\.fq)?(\.fastq)?(\.gz)?$/
+    seqCenter = params.seqCenter ? "--sam-RG ID:${prefix} --sam-RG 'CN:${params.seqCenter}'" : ''
```
[the remark this answers is not in the document]
Comment (ewels) [reply main.nf]: Done 👍

File: main.nf [hunk 7]
```diff
@@ -351,6 +380,7 @@ process bowtie_miRBase_hairpin {
     script:
     index_base = index.toString().tokenize(' ')[0].tokenize('.')[0]
     prefix = reads.toString() - '.mature_unmapped.fq.gz'
+    seqCenter = params.seqCenter ? "--sam-RG ID:${prefix} --sam-RG 'CN:${params.seqCenter}'" : ''
```
[the remark this answers is not in the document]
Comment (ewels) [reply main.nf]: Done 👍

**Comment by drpatelh** [comment]
@ewels @apeltzer @chuan-wang 

How far off is this PR to being merged? We will need to do some major updates to this pipeline in order to bring it in line with the latest template code. I don't see a `dev` branch either which is why I'm wondering how we should manage this PR.

**Comment by apeltzer** [comment]
Maybe we could change this to merge in a new branch dev and work on it afterwards ?

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by drpatelh** [comment]
Yes. That makes sense. Ill try and update the template code when I get a moment.

**Comment by drpatelh** [comment]
Ive added a `dev` branch and changed the base branch for this PR. 

Ill leave it up to you guys as to how you want to merge/fix this PR because you have requested changes. Also, lint and pipeline tests were failing at the time with Travis CI...

**Comment by lpantano** [comment]
Hi all,

I would really like to work on the pipeline, anything I can do to push this forward and start implementing some other tools and strategies?

Can I address the revisions?

Thanks!

**Comment by drpatelh** [comment]
That would be great @lpantano ! Sorry, I haven't had time to update the template code.

I think we should maybe just merge this PR as it stands and then you can develop the pipeline from that point on to get everything working.

**Comment by lpantano** [comment]
I am happy with that, maybe that works better, so +1 from me.

**Comment by apeltzer** [comment]
Go ahead - glad if someone takes care @lpantano !

**Comment by apeltzer** [comment]
Ok I just merged this to `dev` - please create a branch off it and then improve the open issues / points on it to make this work / follow the latest template :-)

**Comment by lpantano** [comment]
perfect! thanks.

**Pull request merged**